### PR TITLE
First pass at a solution

### PR DESCRIPTION
**Pull request by vaughan99**

### CODE REVIEW

File: lib/providers/location.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * This is a provider to find a lat-lon for a given address.
+ * This helps pin point weather at a specific location.
+ */
+
+const requestPromise = require('request-promise-native');
```
Comment (ziplokk1) [inline lib/providers/location.js:6]: What's the difference between request-promise and request-promise-native?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+
+const weatherProvider = require('./lib/providers/weather');
+const routes = require('./lib/routes');
+
+const app = express();
+routes(app);
```
Comment (ziplokk1) [inline index.js:7]: Why use this pattern instead of a router?

File: lib/providers/location.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const locationProvider = require('./location');
+const requestPromiseNative = require('request-promise-native');
+
+jest.mock('request-promise-native');
+
+test('should get latitude/longitude from user data', async () => {
```
Comment (ziplokk1) [inline lib/providers/location.test.js:6]: Beautiful test

File: lib/providers/user.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    // Decrypt the PII.
+    if (user) {
+        return Object.assign(
+            {},
+            user,
+            { pii: JSON.parse(piiCrypto.decryptPii(user.pii)) }
+        );
+    }
+
+    return null;
```
Comment (ziplokk1) [inline lib/providers/user.js:36]: What status code will this return?

**Comment by mtdefelice** [comment]
Thanks! Got it running locally and will sync with Mark on our questions that will frame the technical interview. Look forward to chatting more with you.

**Comment by haliekoehler** [comment]
Thanks Chris! Runs locally without any problem :)

File: lib/providers/location.js (hunk 1 above)
Comment (vaughan99) [reply lib/providers/location.js:6]: The original `request-promise` library was written to bluebird specs. The `request-promise-native` library was rewritten to use native es6 promises.

File: index.js (hunk 2 above)
Comment (vaughan99) [reply index.js:7]: Good question. I should have used a router. :)

File: lib/providers/user.js (hunk 4 above)
Comment (vaughan99) [reply lib/providers/user.js:36]: If I understand your question correctly, the middleware will detect and return 404 if the user is null.

File: index.js (hunk 2 above)
Comment (vaughan99) [reply index.js:7]: Would you like me to update the PR using a router?

Comment (ziplokk1) [reply index.js:7]: Oh, no, not at all. I was just curious.

**Comment by ziplokk1** [comment]
Hey Chris,

We'd like to schedule a follow up interview to chat with you about the
project. What does your availability look like?

**Comment by ziplokk1** [comment]
I'll be reaching out to you via email from here out.

**Pull request closed without merging**